infra/utils.sh

```diff
@@ -0,0 +1,173 @@
+        --max-pods-per-node "${MAX_PODS_PER_NODE}" \
+        --no-issue-client-certificate \
+        --metadata disable-legacy-endpoints=true \
+        --service-account "$(svc_acct_for "${PROJECT}" "${SVC_ACCOUNT}")" \
+        --security-group "gke-security-groups@kubernetes.io" \
+        --enable-stackdriver-kubernetes \
+        --resource-usage-bigquery-dataset "${DATASET_NAME}" \
+        --quiet
+
+    if ! gcloud beta container clusters describe test-cluster --format="value(resourceUsageExportConfig)" >/dev/null 2>&1; then
+        gcloud beta container clusters update "${NAME}" --region "${REGION}" --resource-usage-bigquery-dataset "${DATASET_NAME}"
+    fi
+}
+
+# Sets up Cloud NAT for the network.
+# Assumed vars:
+#   NETWORK_PROJECT
+#   REGION
+#   NETWORK
+function create-cloud-nat-router() {
+    if [ $# -lt 1 ] && [ $# -gt 2 ] && [ -z "$1" ]; then
+        printf "create-cloud-nat-router(project, network, region) requires 2 or 3 arguments" >&2
+        return 1
+    fi
+    local -r project="${1}"
+    local -r network="${2}"
+    local router_name="$network-nat-router"
+    local nat_name="$network-nat-config"
+    local region=${region:=us-central1}
+
+    if ! gcloud --project "${project}" compute routers describe "${router_name}" &>/dev/null; then
+        gcloud --project "${project}" compute routers create "${router_name}" \
+            --region "${region}" \
+            --network "${network}"
+    else
+        printf "Found existing CloudNAT router : %s.\n" "${router_name}"
+    fi
+
+    if ! gcloud --project "${project}" compute routers nats describe "${nat_name}" --router "${router_name}" &>/dev/null; then
+        gcloud --project "${project}" compute routers nats create "${nat_name}" \
+            --router-region "${region}" \
+            --router "${router_name}" \
+            --nat-all-subnet-ip-ranges \
+            --auto-allocate-nat-external-ips
+    else
+        printf "Found existing NAT route : %s.\n" "${nat_name}"
+    fi
+}
+
+function create-network() {
+
+    local project="$1"
+    local network_name="$2"
+    if ! gcloud --project "${project}" compute networks describe "${network_name}" &>/dev/null; then
+        gcloud --project "${project}" compute networks create "${network_name}" --subnet-mode custom
+    else
+        printf "Found existing network: %s.\n" "${network_name}"
+    fi
+}
+
+function create-subnet() {
+
+    local project="$1"
+    local subnet_name="$2"
+    if ! gcloud --project "${project}" compute networks subnets describe "${subnet_name}-custom-subnet" >/dev/null 2>&1; then
+        gcloud --project "${project}" compute networks subnets create "${subnet_name}-custom-subnet" \
+            --network "${NAME}" \
+            --region "${REGION}" \
+            --range "${NODE_IP_RANGE}" \
+            --enable-private-ip-google-access
+    else
+        printf "Found existing subnet: %s.\n" "${subnet_name}"
+    fi
+}
+
+function ensure-service-account() {
```

Comment:
Is there a reason we are creating a specific service account for the cluster?

Reply:
The default service account used has a `Editor` role. This has more permissions than are required to run the GKE clusters. Also I'll prefer to have a service account per cluster instead to use a default service account or just one service account.  WDYT ?

Reply:
Yeah, actually you're right, a service account is better.